src/Http/Http.Abstractions/src/Routing/EndpointMetadataCollection.cs

```diff
@@ -151,8 +151,10 @@ public EndpointMetadataCollection(params object[] items)
         /// </summary>
         public struct Enumerator : IEnumerator<object?>
         {
+#pragma warning disable IDE0044
             // Intentionally not readonly to prevent defensive struct copies
-            private readonly object[] _items;
+            private object[] _items;
```

Comment:
@pranavkm This got changed by #34215 but the comment suggests it shouldn't have. I put it back when resolving merge conflicts.